File: reference/fud-ai/android/app/src/main/java/com/apoorvdarshan/calorietracker/models/ChatMessage.kt
package com.apoorvdarshan.calorietracker.models

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable
import java.time.Instant
import java.util.UUID

@Serializable
data class ChatMessage(
    @Serializable(with = UuidSerializer::class)
    val id: UUID = UUID.randomUUID(),
    val role: Role,
    val content: String,
    val attachmentImageBase64: String? = null,
    @Serializable(with = InstantSerializer::class)
    val timestamp: Instant = Instant.now()
) {
    @Serializable
    enum class Role {
        @SerialName("user") USER,
        @SerialName("assistant") ASSISTANT
    }
}

File: reference/fud-ai/ios/calorietracker/Models/ChatMessage.swift
import Foundation

struct ChatMessage: Identifiable, Codable, Equatable {
    enum Role: String, Codable {
        case user
        case assistant
    }

    let id: UUID
    let role: Role
    let content: String
    let timestamp: Date
    let attachmentImageData: Data?

    init(
        id: UUID = UUID(),
        role: Role,
        content: String,
        timestamp: Date = .now,
        attachmentImageData: Data? = nil
    ) {
        self.id = id
        self.role = role
        self.content = content
        self.timestamp = timestamp
        self.attachmentImageData = attachmentImageData
    }
}
